### PR TITLE
Add support for various JsDiff text comparison methods

### PR DESCRIPTION
**Pull request by circAssimilate**
This PR is heavily inspired by #31, but differs slightly in it's approach. This is required functionality if we're going to be able to use react-diff-viewer in our product. Today, the default text diff comparison looks like this:

Screen Shot 2019-11-18 at 11 58 34 PM

But it would be more desirable for the implementer to be able to specify which [JsDiff method](https://github.com/kpdecker/jsdiff/tree/v4.0.1#api) is used, creating a more desirable diff in cases like this:

Screen Shot 2019-11-18 at 11 58 43 PM

Because the proposed `jsDiffCompareMethod` prop has a default value of `diffChars`, this would be a backwards compatible and a non-breaking change. `disableWordDiff` would still exist as a different prop and we'd simply return early if an unrecognized method name was passed.

In summary, this PR will:
- Update the example app with the text diff method options
- Add readme documentation
- Add tests for the new functionality 

Some other improvements worth considering in this or subsequent PRs:
- Allowing for options documented in https://github.com/kpdecker/jsdiff/tree/v4.0.1#api

**Run this locally:**
_(NOTE: this node version works for me but someone should probably add `.nvmrc` with ideal node version)_
```
# pull this branch / PR
nvm install 12.13.0
nvm use 12.13.0
yarn
yarn build:watch 

# In separate shell
nvm use 12.13.0
yarn start:examples
```

Totally open to feedback and very interested in helping push this over the line!

### CODE REVIEW

File: src/compute-lines.ts [hunk 1]
```diff
@@ -60,14 +70,15 @@ const constructLines = (value: string): string[] => {
...
  * @param oldValue Old word in the line.
  * @param newValue New word in the line.
+ * @param jsDiffCompareMethod JsDiff text diff method from https://github.com/kpdecker/jsdiff/tree/v4.0.1#api
  */
-const computeWordDiff = (oldValue: string, newValue: string): WordDiffInformation => {
-  const diffArray = diff
-    .diffChars(oldValue, newValue);
-  const wordDiff: WordDiffInformation = {
+const computeDiff = (oldValue: string, newValue: string, jsDiffCompareMethod: string): ComputedDiffInformation => {
+  const diffArray = diff[jsDiffCompareMethod](oldValue, newValue);
```
Comment (circAssimilate) [inline src/compute-lines.ts]: I'm hitting a wall trying to solve https://travis-ci.com/praneshr/react-diff-viewer/builds/137136350 this typescript error. Would appreciate any input / direction.

Comment (circAssimilate) [reply src/compute-lines.ts]: Got this and another compile error sorted out by defining how the diff import works and by more explicitly typing the destructured `DiffInformation` object

**Comment by praneshr** [comment]
Hi @circAssimilate, thanks for this. This is really useful. I'll take a look at this over the weekend. Thanks again.

File: src/compute-lines.ts [hunk 2]
```diff
@@ -60,39 +74,40 @@ const constructLines = (value: string): string[] => {
...
-  const wordDiff: WordDiffInformation = {
+const computeDiff = (oldValue: string, newValue: string, jsDiffCompareMethod: string): ComputedDiffInformation => {
+  const diffArray = jsDiff[jsDiffCompareMethod](oldValue, newValue);
+  const computedDiff: ComputedDiffInformation = {
     left: [],
     right: [],
   };
   diffArray
-    .forEach(({ added, removed, value }): DiffInformation => {
+    .forEach(({ added, removed, value }: DiffInformation): DiffInformation => {
```
Comment (praneshr) [inline src/compute-lines.ts]: In `({ added, removed, value }: DiffInformation)`, the type information here is wrong. It should be the type returned by `jsDiff[jsDiffCompareMethod](oldValue, newValue)`

File: src/index.tsx [hunk 3]
```diff
@@ -29,6 +30,8 @@ export interface ReactDiffViewerProps {
   splitView?: boolean;
   // Enable/Disable word diff.
   disableWordDiff?: boolean;
+  // JsDiff text diff method from https://github.com/kpdecker/jsdiff/tree/v4.0.1#api
+  jsDiffCompareMethod?: string;
```
Comment (praneshr) [inline src/index.tsx]: Can we just call it `compareMethod`?

**Comment by circAssimilate** [comment]
Thanks for the review, @praneshr. I'll aim to get to this today!

File: src/compute-lines.ts (hunk 2 above)
Comment (circAssimilate) [reply src/compute-lines.ts]: Great catch. Ended up defining an interface up above. Let me know what you think!

**Comment by circAssimilate** [comment]
@praneshr, all ready for another review when you get the chance. 🎩👌

**Comment by circAssimilate** [comment]
@praneshr, out of curiosity, do you have a rough timeline of when the 3.0.0 release will happen? We're doing some internal planning and I was hoping to communicate some expected timelines.

**Comment by praneshr** [comment]
@circAssimilate I'm so sorry for the delay. v3.0 will be out by Dec 8th. I'll merge your PR with my changes. Till then, you can use the forked module.

**Comment by circAssimilate** [comment]
Sounds good, thank you!

**Comment by praneshr** [comment]
@circAssimilate Thanks for the contribution. Merging this to master. It will be released with v3

**Comment by circAssimilate** [comment]
Thank you for the support and review, @praneshr!
